python_ta/transforms/setendings.py

```diff
@@ -191,6 +295,60 @@ def discover_nodes(node):
     logging.info(message)
 
 
+def fix_slice(source_code):
+    """
+    The Slice node column positions are mostly set properly when it has (Const) children, all it needs is to consume leading/trailing whitespace.
+
+    The main problem is when Slice node doesn't have children. E.g "[:]", "[::]", "[:][:]", or even "[::][::]", yikes! The existing col_offset of the slice node is set improperly to 0. And the end_col_offset is also wrong. We fix it here.
+    """
+    def _consume_subscripts(node):
+
+        if not node.last_child():
+            if not hasattr(node, 'end_lineno'): set_without_children(node)
+
+            line_i = node.parent.fromlineno - 1  # 1-based
+            char_i = node.parent.col_offset  # 0-based
+
+            # To solve the problem created by variations of "a[:][:]",
+            # all sibling Subscript node characters to consume.
+            sibling_buffer = ''
```

Comment:
Sadly (because it would have been very nice) this will not work to set Slice() in the cases where the `value` node is a `Subscript()` node. Because the ending locations of `Subscript()` nodes are set from the child nodes. This sequence should explain what I am trying to say:
-- Subscript() ending locations are set from child node,
-- Slice() is a child of Subscript(),
-- Slice() needs to be set before Subscript(),
-- therefore the ending position of the parent's `value` node (in the cases where the value node is a Subscript()) cannot be used to set the Slice() ending locations..

```
...
┃  ┃  ┃━ Subscript(), name: None,  value: Subscript()                 💥💥💥
┃  ┃  ┃  fromlineno:       15 ... end_lineno:       15
┃  ┃  ┃  col_offset:        0 ... end_col_offset:    7
┃  ┃  ┃  >>>n[:][:]    <============== cannot use end_col_offset here though :-(
┃  ┃  ┃  >>>n[:][:]
┃  ┃  ┃  ┃━ Subscript(), name: None,  value: Name(n)                 💥💥💥
┃  ┃  ┃  ┃  fromlineno:       15 ... end_lineno:       15
┃  ┃  ┃  ┃  col_offset:        0 ... end_col_offset:    4
┃  ┃  ┃  ┃  >>>n[:]
┃  ┃  ┃  ┃  >>>n[:]
┃  ┃  ┃  ┃  ┃━ Name(n), name: n,  value: None
┃  ┃  ┃  ┃  ┃  fromlineno:       15 ... end_lineno:       15
┃  ┃  ┃  ┃  ┃  col_offset:        0 ... end_col_offset:    1
┃  ┃  ┃  ┃  ┃  >>>n
┃  ┃  ┃  ┃  ┃  >>>n
┃  ┃  ┃  ┃  ┃━  Slice(), name: None,  value: None
┃  ┃  ┃  ┃  ┃  fromlineno:       15 ... end_lineno:       15
┃  ┃  ┃  ┃  ┃  col_offset:        2 ... end_col_offset:    3
┃  ┃  ┃  ┃  ┃  >>>:
┃  ┃  ┃  ┃  ┃  >>>:
┃  ┃  ┃  ┃━ Slice(), name: None,  value: None                 💥💥💥
┃  ┃  ┃  ┃  fromlineno:       15 ... end_lineno:       15
┃  ┃  ┃  ┃  col_offset:        5 ... end_col_offset:    6
┃  ┃  ┃  ┃  >>>:    <============== want to set the col_offset, end_col_offset.
┃  ┃  ┃  ┃  >>>:
...
```